jmespath-vertx/src/main/java/io/burt/jmespath/vertx/VertxRuntime.java

```diff
@@ -0,0 +1,213 @@
+package io.burt.jmespath.vertx;
+
+import io.burt.jmespath.BaseRuntime;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.RuntimeConfiguration;
+import io.burt.jmespath.jcf.JcfRuntime;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.util.*;
+
+import static io.burt.jmespath.JmesPathType.*;
+
+public class VertxRuntime extends JcfRuntime {
+    public VertxRuntime() {
+        this(RuntimeConfiguration.defaultConfiguration());
+    }
+
+    public VertxRuntime(RuntimeConfiguration configuration) {
+        super(configuration);
+    }
+
+    @Override
+    public Object parseString(String str) {
+        return Json.decodeValue(str);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<Object> toList(Object value) {
+        if (value instanceof JsonArray) {
+            final JsonArray ja = (JsonArray)value;
+            return new AbstractList<Object>() {
+
+                @Override
+                public int size() {
+                    return ja.size();
+                }
+
+                @Override
+                public Object get(int i) {
+                    return ja.getValue(i);
+                }
+            };
+        }
+        if (value instanceof JsonObject) {
+            final JsonObject jo=(JsonObject)value;
+            return new AbstractList<Object>() {
+                // Keep a cache of the ordered list if random access comes up
```

Comment:
It would be premature optimization unless the consumer just wants size.  Otherwise, if they want an iterator or to get an individual element, it's worth the O(1) expense.